Mathlib/Init/Algebra/Classes.lean

```diff
@@ -67,219 +67,219 @@ set_option autoImplicit true
 universe u v
 
 -- porting note: removed `outParam`
-class IsSymmOp (α : Type u) (β : Type v) (op : α → α → β) : Prop where
+class IsSymmOp (α : Sort u) (β : Sort v) (op : α → α → β) : Prop where
```

Comment:
Can you please record a summary (maybe as library note?) that motivates why these classes should be defined on `Sort`s instead of `Type`s?
It's not clear to me why we would want to allow Props here...

Reply:
One thing I had to fix in #10335 was that the `IsTrans` defined in this file gained higher priority, and then Lean tried to apply it to `Eq` in calc blocks and failed because of universe issues, as `IsTrans` was only defined on `Type u` and it was trying to apply it to `Prop`. One fix was to extend the `IsTrans` to `Sort` instead of `Type`, and then I thought that it would make sense to have all these definitions on `Sort` as it makes universe unification a bit easier, for essentially no cost. A better fix, IMHO, would be to just remove this file as we don't really use it.

Reply:
We can also discard the PR, it's not as if it's really important.

Reply:
I think your motivation is valid. Let's just merge this.

bors merge